packages/core/src/auth/auth.ts

```diff
@@ -223,6 +245,8 @@ export class Auth implements AuthService, ConnectionManager {
         this.#onDidChangeActiveConnection.fire(conn)
         await this.store.setCurrentProfileId(id)
 
+        await setContext('aws.isInternalUser', this.isInternalAmazonUser())
```

Comment:
I was wondering if `forgetConnection` and `expireConnection` should also un-set this. But it's probably useful to keep this hint around, once it was deduced. And in that case, should this flag be stored in globalState? Then we don't need to wait until a valid connection next time.

Reply:
I don't think it hurts to set this back to false on `forgetConnection` and `expireConnection`.

Some testing that I did made me doubt how `globalState` worked. Using `globalState.update` for this didn't work to enable the feature for internal users. `setContext`, however, worked. Are global keys detectable from package.json after being set?

Reply:
> Are global keys detectable from package.json after being set?

No

Reply:
Oh okay well that would explain it. We need this to be set with `setContext`